### PR TITLE
fix: Change to Set vs FrozenSet and thread the same set through

### PR DESCRIPTION
**Pull request by tmc**
This improves the performance dramatically for large type hierarchies.

Addresses #950

### CODE REVIEW

**Comment by software-dov** [comment]
Thanks so much for making these changes! Performance on big, complex APIs has been a pain point for a while.
What APIs have you tested on? Do you have any before and after benchmark data? I'm a little bit concerned about changing `frozenset` to `set` because of the risk of mutating the caller's parameters. If you can show that the performance gained from that is substantial it would put my mind at ease.

**Comment by vam-google** [comment]
@tmc PTAL

**Comment by tmc** [comment]
@vam-google any example that reference the protobuf types from https://github.com/google/fhir will not effectively finish without a change such as this one.

**Comment by parthea** [comment]
@tmc Please could you address the review comment and feedback from software-dov?

**Comment by tmc** [comment]
@atulep @parthea PTAL.

**Comment by parthea** [comment]
Converting to draft as presubmits are failing

**Comment by tmc** [comment]
@parthea PTAL

File: gapic/schema/wrappers.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 Documentation is consistently at ``{thing}.meta.doc``.
 """
 
+import sys
```
Comment (parthea) [inline gapic/schema/wrappers.py:30]:
```suggestion
```

**Review by parthea — DISMISSED** [review]

**Review by parthea — APPROVED** [review]

**Pull request merged**